build/source/engine/sunGeomtry.f90

```diff
@@ -94,7 +95,14 @@ SUBROUTINE CLRSKY_RAD(MONTH,DAY,HOUR,DT,SLOPE,AZI,LAT,HRI,COSZEN)
  ! Calculate latitude "adjustment" for ground slope, aspect and latitude (LP = LAT1 for level surface)
  LP=ASIN(SIN(SLOPE1)*COS(AZI1)*COS(LAT1) + COS(SLOPE1)*SIN(LAT1)) ! angle between solar rays and surface (tilted) ??
  ! Calculate time of sunrise/sunset on level surface as radians from noon
- TD=ACOS(-TAN(LAT1)*TAN(D))
+ ! Account for low/high latitude locations, where there might not be a sunrise/sunset time on a given day
```

Comment:
I think you mean just "high latitude locations", since low latitudes are at the equator.

Reply:
Changed as requested in the most recent commit

Reply:
Code can be compiled with these changes and runs without crashing for locations up to 75 degrees latitude. SWE simulations don't look strange but we've not done any tests that compare SUMMA output to known solutions.

Reply:
In this case, the tests should include:
a) no change (bit-for-bit) identical for situations where you do not have 24 hour sunlight (first: do no harm)
b) the change handles the case as expected for cases where you have 24 hour sunlight

Looks like you did some of these runs already, but the tests should be a bit stricter than "does not crash". Even you can analyze your tests a bit further and report on a) and b) then I can go ahead and merge. Note that you can attach figures etc to the discussion in github (so not as part of a commit, but simply here).

Reply:
Here are the requested checks, comparing HRU 275 (before and after code change to check that no changes were introduced in cases that worked fine before), and HRU 274 (after code change only, to see whether the changes resolved the crash and whether the simulations look reasonable). Because HRU 274 and 275 are adjacent but still cover an area of approximately 300 km^2 each, I expect that their fluxes should be roughly similar in seasonal patterns but do not expect them to be identical.

Left column contains SWE, Q and ET outputs for HRU 275. Simulations before and after code change are identical.

Right column contains SWE, Q and ET outputs for adjacent HRU 274. After code changes the three fluxes are comparable to those of HRU 275.

![sunGeom_changes_assessment](https://user-images.githubusercontent.com/45757529/85176598-696e4080-b237-11ea-89fa-86205adafd56.png)

Reply:
Hi @wknoben, thanks for that. Almost there. I think this nicely shows that 275 after the fix is similar to 274. The one thing left is to show that for 274 the results are actually bit-for-bit the same. You would not really be able to tell this from a plot. It means that the output file are identical (other than global attributes). There are programs (like nccmp) that you can use to ascertain this. If this makes no sense, please check with @martynpclark. After that test I can merge the PR.

Reply:
Right, I ran nccmp on the timestep and day output files for the original and the new code as follows:
nccmp -lmbdgS --precision=%x  originalCode_na_grid_era5_output_G00275-00275_timestep.nc newCode_na_grid_era5_output_G00275-00275_timestep.nc > nccmp_timestepOutput_verbose.txt

As far as I can tell it all comes up clean.

**Daily output comparison as saved to file:**
2020-06-24 11:54:36.486430 -0600 [0;32mINFO[0m nccmp.c:140 Command-line options parsed.
2020-06-24 11:54:36.486535 -0600 [0;32mINFO[0m nccmp.c:44 Opening input files.
2020-06-24 11:54:36.486541 -0600 [0;32mINFO[0m nccmp_state.c:149 Opening originalCode_na_grid_era5_output_G00275-00275_day.nc
2020-06-24 11:54:36.491776 -0600 [0;32mINFO[0m nccmp_state.c:149 Opening newCode_na_grid_era5_output_G00275-00275_day.nc
2020-06-24 11:54:36.495633 -0600 [0;32mINFO[0m nccmp.c:82 Comparing file formats.
2020-06-24 11:54:36.495646 -0600 [0;32mINFO[0m nccmp.c:96 Comparing groups.
2020-06-24 11:54:36.495698 -0600 [0;32mINFO[0m nccmp_group.c:53 Comparing groups "/" and "/".
2020-06-24 11:54:36.495706 -0600 [0;32mINFO[0m nccmp_metadata.c:1566 Comparing global attributes.
2020-06-24 11:54:36.496614 -0600 [0;32mINFO[0m nccmp_group.c:135 Group comparison complete.

**Timestep output comparison as saved to file:**
2020-06-24 11:55:40.743457 -0600 [0;32mINFO[0m nccmp.c:140 Command-line options parsed.
2020-06-24 11:55:40.743559 -0600 [0;32mINFO[0m nccmp.c:44 Opening input files.
2020-06-24 11:55:40.743564 -0600 [0;32mINFO[0m nccmp_state.c:149 Opening originalCode_na_grid_era5_output_G00275-00275_timestep.nc
2020-06-24 11:55:40.839577 -0600 [0;32mINFO[0m nccmp_state.c:149 Opening newCode_na_grid_era5_output_G00275-00275_timestep.nc
2020-06-24 11:55:40.896651 -0600 [0;32mINFO[0m nccmp.c:82 Comparing file formats.
2020-06-24 11:55:40.896670 -0600 [0;32mINFO[0m nccmp.c:96 Comparing groups.
2020-06-24 11:55:40.896743 -0600 [0;32mINFO[0m nccmp_group.c:53 Comparing groups "/" and "/".
2020-06-24 11:55:40.896752 -0600 [0;32mINFO[0m nccmp_metadata.c:1566 Comparing global attributes.
2020-06-24 11:55:40.897841 -0600 [0;32mINFO[0m nccmp_group.c:135 Group comparison complete.